Dockerfile.dev

```diff
@@ -1,4 +1,4 @@
-FROM mcr.microsoft.com/vscode/devcontainers/python:0-3.9
```

Comment:
Our minimum supported Python version is 3.9, hence we stick with 3.9 at this moment to prevent 3.10 exclusive features to sneak in.

Reply:
Ah my bad - it seems like the actual solution to the linter error `Alternative syntax for unions requires Python 3.10` is [documented here](https://community.home-assistant.io/t/alternative-syntax-for-unions-requires-python-3-10-or-newer/402411/8).